libpod/image/image.go

```diff
@@ -210,8 +194,19 @@ func (ir *Runtime) SaveImages(ctx context.Context, namesOrIDs []string, format s
 		writer = os.Stderr
 	}
 
+	policyContext, err := getPolicyContext(sys)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := policyContext.Destroy(); err != nil {
+			logrus.Errorf("failed to destroy policy context: %q", err)
+		}
+	}()
+
 	// extend an image with additional tags
 	type imageData struct {
+		destNames []string
```

Comment:
If “extend an image”, the `*Image` should probably remain first.

I’m not 100% happy with the two duplicate name arrays, but I also can’t see much of an alternative. (Moving the docker-archive tag parsing into the docker-archive specific loop would probably add too much complexity.)